src/main/java/org/tikv/common/policy/RetryPolicy.java

```diff
@@ -67,7 +68,16 @@ private void rethrowNotRecoverableException(Exception e) {
     }
   }
 
-  public RespT callWithRetry(Callable<RespT> proc, String methodName) {
+  public RespT callWithRetry(Callable<RespT> proc, String methodName, BackOffer backOffer) {
+    SlowLogTask slowLogTask = backOffer.slowLogStart("callWithRetry " + methodName);
+    try {
```

Comment:
if the exception is happened and is not caught, will the request fail? do we still need to trace and record the slow operations in this scenario?

Reply:
if exception happends, the request will fail, we do trace and recode the operation by the code:

```
finally {
      slowLogSpan.end();
    }
```

Reply:
Do we need to investigate the slow log reason when the request is failed?

Reply:
add in line 80 will increase the code complexity.

Reply:
you can put this slow log tracer together with `callWithRetryTimer`:

```java
Histogram.Timer callWithRetryTimer = CALL_WITH_RETRY_DURATION.labels(methodName).startTimer();
SlowLogSpan slowLogSpan = backOffer.getSlowLog().start("callWithRetry " + methodName);
try {
  // send gRPC request, receive response and handle errors
} finally {
  callWithRetryTimer.observeDuration();
  slowLogSpan.end();
}
```

I think it's much simple because:
- it reduces 1 function (no more this wrapper)
- the two timers put are together, readers can easily understand and maintain the code in the future
- less function call, less exception handling, more efficient

Reply:
sounds good